### PR TITLE
Remove Android NDK20 and Android SDK 32

### PR DESCRIPTION
**Pull request by AGulev**
Our tests have been broken for a month. I tried to find the reason, but I can't figure that out. I don't want to merge this commit until I'll be able to pass the test with it

### CODE REVIEW

File: server/docker-base/Dockerfile [hunk 1]
```diff
@@ -448,16 +393,15 @@ RUN \
...
-    which aarch64-linux-android${ANDROID_64_NDK20_API_VERSION}-clang++ && \
+    ls -la ${ANDROID_SDK_BUILD_TOOLS_PATH_33}/d8 && \
     which ${ANDROID_NDK25_BIN_PATH}/armv7a-linux-androideabi${ANDROID_NDK25_API_VERSION}-clang++ && \
-    which ${ANDROID_NDK25_BIN_PATH}/aarch64-linux-android${ANDROID_64_NDK25_API_VERSION}-clang++ && \
+    which ${ANDROID_NDK25_BIN_PATH}/aarch64-linux-android${ANDROID_64_NDK25_API_VERSION}-clang++
 # check that aapt installed correctly
-    aapt v
+# We specify ANDROID_SDK_BUILD_TOOLS_PATH_* in build_input.yml by setting it the first in PATH
+# We can't check aapt in initial setup anymore
+#   aapt v
```
Comment (AGulev) [inline server/docker-base/Dockerfile:404]: to be able to use `aapt` from different SDKs we have to specify them from `build_input.yml`

File: server/docker-base/Dockerfile [hunk 2]
```diff
@@ -610,7 +554,7 @@ COPY template.umbrella.h ${EXTENSION_UMBRELLAHEADER_TEMPLATE}
 
 ENV ZIG_VERSION=0.11.0
 ENV ZIG_PATH_0_11=${PLATFORMSDK_DIR}/zig-0-11
-ENV ZIG_URL=https://ziglang.org/builds/zig-linux-x86_64-${ZIG_VERSION}-dev.3886+0c1bfe271.tar.xz
+ENV ZIG_URL=https://ziglang.org/download/${ZIG_VERSION}/zig-linux-x86_64-${ZIG_VERSION}.tar.xz
```
Comment (AGulev) [inline server/docker-base/Dockerfile:557]: old link to the dev build isn't available anymore

**Review by JCash — APPROVED** [review]

File: server/docker-base/Dockerfile [hunk 3]
```diff
@@ -349,34 +336,14 @@ ENV ANDROID_SDK_FILENAME_33 android-sdk-linux-android-${ANDROID_SDK_VERSION_33}-
...
-
-
-
-# We must keep two NDKs alive for now, since migrating to the latest (i.e 25b) will break
-# for users that try to build with an older build.yml.
+# We specify it in build_input.yml by setting it the first in PATH for new SDK
+# But at least one SDK should be specified by default in Dockerfile
+ENV PATH ${PATH}:${ANDROID_HOME}/tools:${ANDROID_HOME}/platform-tools:${ANDROID_SDK_BUILD_TOOLS_PATH_33}
+# ANDROID_SDK_VERSION used in Java code, default should be specified in Dockerfile as well:
+ENV ANDROID_SDK_VERSION ${ANDROID_SDK_VERSION_33}
```
Comment (AGulev) [inline server/docker-base/Dockerfile:343]: we use it before sdk downloading

File: server/test-data/sdk/a/defoldsdk/extender/build.yml [hunk 4]
```diff
@@ -148,9 +151,9 @@ platforms:
...
+    linkCmd: '{{env.NDK_CXX}} -fuse-ld={{env.NDK_LD}} -O2 -g -isysroot={{env.SYSROOT}} -static-libstdc++ {{#src}}{{.}} {{/src}} -o {{tgt}} -Wl,-soname=libdmengine.so {{#defines}}-D{{{.}}} {{/defines}} {{#linkFlags}}{{{.}}} {{/linkFlags}} -Wl,--fix-cortex-a8 -Wl,--no-undefined -Wl,-z,noexecstack -landroid -fpic -z text {{#ext.libPaths}}-L{{.}} {{/ext.libPaths}} {{#libPaths}}-L{{.}} {{/libPaths}} -Wl,-Bstatic -Wl,--start-group -lc++_static {{#libs}}-l{{{.}}} {{/libs}} {{#ext.libs}}-l{{.}} {{/ext.libs}} {{#engineLibs}}-l{{.}} {{/engineLibs}} -Wl,--end-group -Wl,-Bdynamic {{#ext.dynamicLibs}}-l{{{.}}} {{/ext.dynamicLibs}} {{#dynamicLibs}}-l{{{.}}} {{/dynamicLibs}} -Wl,--no-undefined -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -shared'
+    libCmd: 'llvm-ar rcs {{tgt}} {{#objs}}{{.}} {{/objs}}'
```
Comment (AGulev) [inline server/test-data/sdk/a/defoldsdk/extender/build.yml:156]: These fixes needed to be  compatible with ndk25

**Comment by AGulev** [comment]
I tested it locally and tests are green now

**Review by JCash — APPROVED** [review]

**Pull request merged**